src/blockstore/keyval.go

```diff
@@ -0,0 +1,117 @@
+//
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package blockstore
+
+// This file implements key/value primitives.
+
+import (
+	"math/rand"
+	"sort"
+	"time"
+)
+
+// The key is 7 least significcant bytes.
+// The highmost byte is reserved for flags.
+const flagsMask = uint64(0xFF) << 56
+const keyMask = ^(uint64(0xFF) << 56)
+const maxKey = ^(uint64(0xFF) << 56)
+
+// This block describes flag bits.
+const (
+	// Defines if an entry is modified/dirty
+	// This is used in the tableManager, so the entry is not evicted.
+	flagDirty = uint8(1) << iota
+	// Defines if an entry is marked for removal.
+	flagRemove
+	// Other bits are reserved.
+)
+
+// Size of a Block can be arbitrary. It is Intended to be 4096 to fit in a 4KB page.
+const blockSize = 4096
+
+type Block [blockSize]byte
+
+// A struct that holds information for a (key, data block) pair.
+// The flags byte would indicate if the keyVal is marked for removal.
+type keyVal struct {
+	key   uint64
+	block *Block
+	flags byte
+}
+
+// A helper function to sort slice of keys, which of are of type uint64.
+func uint64Sort(src []uint64) {
+	var tmpSlice []int
+	tmpSlice = make([]int, len(src), len(src))
+	for i, k := range src {
+		tmpSlice[i] = int(k)
+	}
+	sort.Ints(tmpSlice)
+	for i, k := range tmpSlice {
+		src[i] = uint64(k)
+	}
+}
+
+// Making sure the highest byte of a key is zeroed.
```

Comment:
Use active voice tense in comments: for example "Make sure the highest (most significant) byte of a key is zeroed."

Please change away from -ing verbs wherever possible.

Reply:
Sure, I think this is an old habit. I need to go back to all of them, I will not mark it as resolved, and fix it with the next PR.